docs/book/src/architecture/controllers/machine.md

```diff
@@ -48,14 +48,14 @@ The `status` object **must** have several fields defined:
 
 The `status` object **may** define several fields that do not affect functionality if missing:
 
-* `errorReason` - is a string that explains why an error has occurred, if possible.
-* `errorMessage` - is a string that holds the message contained by the error.
+* `failureReason` - is a string that explains why a fatal error has occurred, if possible.
+* `failureMessage` - is a string that holds the message contained by the error.
 
 Example:
 
 ```yaml
 kind: MyBootstrapProviderConfig
-apiVersion: bootstrap.cluster.x-k8s.io/v1alpha2
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha3
```

Comment:
This section is only relevant after https://github.com/kubernetes-sigs/cluster-api/pull/1800 is merged.